enterprise/server/remote_execution/workspace/workspace_test.go

```diff
@@ -244,3 +246,23 @@ func TestCleanInputsIfNecessary_CleanMatching(t *testing.T) {
 		"expected all KEEPME filePaths (and no others) in the workspace after cleanup",
 	)
 }
+
+func TestManyNewWorkspaces(t *testing.T) {
+	te := testenv.GetTestEnv(t)
+	root := testfs.MakeTempDir(t)
+	var expectedPath *regexp.Regexp
+	if runtime.GOOS == "windows" {
+		expectedPath = regexp.MustCompile(".*" + regexp.QuoteMeta("execroot\\_main"))
+	} else {
+		expectedPath = regexp.MustCompile(".*")
+	}
+	allPaths := make(map[string]struct{})
+	for i := 0; i < 1000; i++ {
+		ws, err := workspace.New(te, root, &workspace.Opts{})
+		require.NoError(t, err)
+		assert.Regexp(t, expectedPath, ws.Path())
+		allPaths[ws.Path()] = struct{}{}
+	}
+	// Check that all paths are unique.
+	assert.Len(t, allPaths, 1000)
+}
```

Comment:
I don't believe we are currently running windows tests on CI (cc @sluongng ) - looks like we're only building the executor binary: https://github.com/buildbuddy-io/buildbuddy/blob/bf374b66e9e123947a429fc3152afc6315836d5c/.github/workflows/build-executor-win.yaml#L53